packages/utils/src/url.ts

```diff
@@ -1,11 +1,14 @@
-type PartialURL = {
-  host?: string;
-  path?: string;
-  protocol?: string;
-  relative?: string;
-  search?: string;
-  hash?: string;
-};
+type PartialURL = Partial<{
+  host: string;
+  path: string;
+  protocol: string;
+  relative: string;
+  search: string;
+  hash: string;
+  urlInstance: URL;
+}>;
+
+const urlRegex = /^(([^:/?#]+):)?(\/\/([^/?#]*))?([^?#]*)(\?([^#]*))?(#(.*))?$/;
```

Comment:
Is there an external reference to the source of this regexp that would make sense to add here? I'm asking because these look daunting (at least to me), but I find that they can usually be a good way to educate, for example maybe by linking to the standard that this is supposed to support.